packages/transformers/js/src/visitors/process.js

```diff
@@ -2,18 +2,30 @@ import * as types from '@babel/types';
 import {morph} from './utils';
 
 export default {
-  MemberExpression(node, {asset, ast, env, isBrowser, isNode}, ancestors) {
+  MemberExpression(
+    node,
+    {asset, ast, env, isBrowser, isNode, replaceEnv},
+    ancestors,
+  ) {
     // Inline environment variables accessed on process.env
     if (!isNode && types.matchesPattern(node.object, 'process.env')) {
       let key = types.toComputedKey(node);
       if (types.isStringLiteral(key)) {
+        let {value} = key;
         // Try using the value from the passed env (either from new Parcel
         // options or from dotenv), and fall back to process.env
-        let prop = env[key.value] ?? process.env[key.value];
+        let prop =
+          value === 'NODE_ENV' ||
+          (process.env.PARCEL_BUILD_ENV !== 'production' &&
+            value === 'PARCEL_BUILD_ENV') ||
```

Comment:
Why allow PARCEL_BUILD_ENV in dev?

Reply:
https://github.com/parcel-bundler/parcel/blob/769c1fcfd98ec4a25270ad5cdbed5f4e9b01d12c/packages/runtimes/js/src/bundle-url.js#L12-L19